### PR TITLE
[Feature] - Award APIs hookup

### PR DESCRIPTION
**Pull request by mkanoria**
Hooking up API's to awards pages

### CODE REVIEW

File: pages/awards/index.js [hunk 1]
```diff
@@ -117,6 +120,66 @@ export default function Performances() {
     return elements;
   };
 
+  async function getAwards() {
+    try {
+      const resp = await axios({
```
Comment (feng-eric) [inline pages/awards/index.js:125]: After my PR gets merged in, it will require an eventID to be specified in the request body

Comment (mkanoria) [reply pages/awards/index.js:132]: Yup, will keep that in mind!

**Comment by mkanoria** [comment]
Force push after rebase - linting might fail (pls ignore)

**Review by feng-eric — CHANGES_REQUESTED** [review]
Left some comments!!

File: pages/api/performances/nominate.js [hunk 2]
```diff
@@ -56,9 +56,6 @@ export default async (req, res) => {
         });
 
         if (nomination) continue;
-      } else if (award.type === 'SCORE_BASED') {
-        // We should not be able to nominate performances for score based awards
```
[the remark this answers is not in the document]
Comment (thepenguinco) [reply pages/api/performances/nominate.js:92]: We're calling nominate API before finalizing a score based award.  Probably not the greatest.  We should definitely make a note of this.

File: pages/awards/[id].js [hunk 3]
```diff
@@ -1,45 +1,113 @@
...
-import PlayIcon from '@assets/play.svg';
+import PlayIcon from '@assets/play.svg'; // Play icon
 import styles from '@styles/pages/AwardDetails.module.scss';
 
 export default function DetailsRoute({ award }) {
-  return award.type === 'Special Award' ? (
-    <AwardDetails />
-  ) : award.type === 'Dance Artistry' ? (
-    <AwardDetails />
+  console.log('award is');
```
Comment (thepenguinco) [inline pages/awards/[id].js:18]: Let's remove console logs on the frontend.  Backend console logs are okay.

File: components/awards/ScoreBasedAwards.js [hunk 4]
```diff
@@ -1,30 +1,72 @@
...
+  const [confirmationModalOpen, setConfirmationModalOpen] = useState(false);
+  const [performanceToFinalize, setPerformanceToFinalize] = useState(-1);
+
+  // Get peformances on load
+  useEffect(() => {
+    getPerformances();
+  }, [event]);
+
+  useEffect(() => {
+    console.log(performances);
```
Comment (thepenguinco) [inline components/awards/ScoreBasedAwards.js:50]: Remove console logs on the frontend

File: components/awards/ScoreBasedAwards.js [hunk 5]
```diff
@@ -1,30 +1,72 @@
...
 import DancerYellowBlue from '@assets/dancer-yellow-blue.svg'; // Jumping Dancer SVG
-import styles from '@styles/pages/Performances.module.scss'; // Page styles
-
-// Temp constants
-import events, { columns } from '../../data/mockParticipants';
+import styles from '@styles/pages/Awards.module.scss'; // Page styles
+import { formatPerformances } from '@utils/performances'; // Format performances util
+import AwardPill from '@components/awards/FinalizePill.js'; // Award Finalize Pill
 
 const PAGE_SIZE = 8; // Rows per page
```
Comment (thepenguinco) [inline components/awards/ScoreBasedAwards.js:21]: For performance page I think we're showing 20 rows per page.  Maybe we can change this to 20 as well here?

File: pages/awards/index.js [hunk 6]
```diff
@@ -1,64 +1,87 @@
...
 import DancerRedJump from '@assets/dancer-red-jump.svg'; // Jumping Dancer SVG
 import DancerYellowBlue from '@assets/dancer-yellow-blue.svg'; // Jumping Dancer SVG
-import styles from '@styles/pages/Performances.module.scss'; // Page styles
+import styles from '@styles/pages/Awards.module.scss'; // Page styles
 
-// Temp constants
-import data, { columns } from '../../data/mockAwards';
-
-const PAGE_SIZE = 6; // Rows per page
+const PAGE_SIZE = 15; // Rows per page
```
Comment (thepenguinco) [inline pages/awards/index.js:26]: Again, page rows is 20 on performances, do we want to keep it the same?

File: pages/awards/index.js [hunk 7]
```diff
@@ -1,64 +1,87 @@
...
   return Object.keys(options).filter(option => options[option].selected);
 };
 
 // Remove key from object (returns new object)
-const removeKeyFromObject = (object, key) => {
-  // eslint-disable-next-line no-unused-vars
-  const { [key]: _, ...rest } = object;
-  return rest;
-};
+// const removeKeyFromObject = (object, key) => {
```
Comment (thepenguinco) [inline pages/awards/index.js:34]: Do we still need this commented code?

**Review by thepenguinco — APPROVED** [review]
Just a few small things, otherwise LGTM.

File: pages/awards/[id].js [hunk 8]
```diff
@@ -1,45 +1,113 @@
...
+      router.push('/awards');
+    } catch {
+      // Empty catch block
+    }
+    setLoading(false);
+  }
+
+  async function unfinalizeAward() {
+    setLoading(true);
+    console.log(performanceToFinalize);
```
Comment (feng-eric) [inline pages/awards/[id].js:85]: Remove later

File: pages/api/performances/nominate.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (feng-eric) [reply pages/api/performances/nominate.js:92]: Should this part be there to stop us from calling nominate API on score-based awards?

**Review by feng-eric — APPROVED** [review]
Other than the comments I had, it looks good!!

File: pages/api/awards/get.js [hunk 9]
```diff
@@ -39,7 +40,16 @@ export const getAward = async filter => {
     include: {
       awards_performances: {
         include: {
-          performances: true,
+          performances: {
+            include: {
+              adjudications: {
+                include: {
+                  user: true,
```
[the remark this answers is not in the document]
Comment (mkanoria) [reply pages/api/awards/get.js:47]: If I understand correctly, this `user` means the user that added the adjudication, whereas the other is the user which possibly nominated for the award

File: pages/awards/index.js (hunk 7 above)
Comment (mkanoria) [reply pages/awards/index.js:34]: This is for the filtering, I'll un-comment in my next change so figure might leave it in?

**Pull request merged**